services/player/youtube-player.js

```diff
@@ -69,13 +71,17 @@ module.exports = class YoutubePlayer extends Player
 
         let track = this._getTrack(queue);
         let trackTitle = track.title.replace(/[ *?!]/gi, '_') + '.mp3';
+        const hash = crypto.createHash('sha256');
+        hash.update(trackTitle);
+        const trackTitleHash = hash.digest('hex');
         let filePath = `${YoutubePlayer.DOWNLOAD_DIR()}`;
-        if (fs.existsSync(`${filePath}\\${trackTitle}`) === false) {
+        const pathAndFilename = path.join(filePath, trackTitleHash);
```

Comment:
I think trackPath would be more self descriptive var name, what do you think?

Reply:
Given that ```filePath``` already exists describing the full directory where the tracks are stored, I think that ```trackPath``` would be very confusing.

If you changed ```filePath``` to ```trackDirectory```, then something like ```fullTrackPath``` would make more sense and wouldn't be so confusing.

Reply:
Agreed, trackDirectory naming would make it easier to read in that regard.